languagetool-language-modules/pt/src/main/java/org/languagetool/rules/pt/BrazilianToponymFilter.java

```diff
@@ -0,0 +1,49 @@
+/* LanguageTool, a natural language style checker
+ * Copyright (C) 2012 Jaume Ortolà i Font
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+package org.languagetool.rules.pt;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.rules.RuleMatch;
+import org.languagetool.rules.patterns.RegexRuleFilter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+
+public class BrazilianToponymFilter extends RegexRuleFilter {
+  @Override
+  public RuleMatch acceptRuleMatch(RuleMatch match, Map<String, String> arguments, AnalyzedSentence sentence, Matcher matcher) {
+    String toponym = matcher.group(1);
+    String underlined = matcher.group(2);
+    String state = matcher.group(3);
+
+    // TODO: read this from user options or something...
+    String suggestion = "–" + state;
+    if (suggestion.equals(underlined)) {
+      return null;
+    }
+    BrazilianToponymMap map = new BrazilianToponymMap();
```

Comment:
Would it make sense to make this a static constant? We maybe don't want to re-initialize the object every time there's a potential match.

Reply:
Yes, we should definitely do that. I was certain I had, but I think I moved it around a bit and forgot to put it back.

Reply:
done